src/main/webapp/app/exercises/shared/result/result-detail.component.ts

```diff
@@ -691,5 +713,23 @@ export class ResultDetailComponent implements OnInit {
         if (this.showOnlyNegativeFeedback || this.showOnlyPositiveFeedback) {
             this.filteredFeedbackList = this.filteredFeedbackList.filter(filterPredicate);
         }
+
+        this.countFeedbacks();
+    }
+
+    private countFeedbacks() {
+        const testCaseList = this.filteredFeedbackList.filter((feedback) => feedback.type === FeedbackItemType.Test);
+        if (this.numberOfAggregatedTestCases && (this.showOnlyPositiveFeedback || !this.showOnlyNegativeFeedback)) {
+            this.testCaseCount = testCaseList.length + this.numberOfAggregatedTestCases - 1;
+            this.passedTestCaseCount = testCaseList.filter((feedback) => feedback.positive).length + this.numberOfAggregatedTestCases - 1;
+        } else {
+            this.testCaseCount = testCaseList.length;
+            this.passedTestCaseCount = testCaseList.filter((feedback) => feedback.positive).length;
+        }
```

Comment:
To me this is not intuitive why there is such a distinction. Could you please add some comments?

For example, why the `-1`? Both are counts so there should be no need for subtraction. If there is need for that, why?
Next: How do the two variables `this.showOnlyPositiveFeedback` and `this.showOnlyNegativeFeedback` play in the calculation. We filter for positive feedback either way and `this.numberOfAggregatedTestCases` is either 0 or contains a number of positive cases.

At least that's what I can see. So some educating documentation would be very much appreciated.